subsys/emds/emds_flash.c

```diff
@@ -366,12 +367,15 @@ static int ate_last_recover(struct emds_fs *fs)
 static int old_entries_invalidate(struct emds_fs *fs)
 {
 	int rc = 0;
-	uint8_t inval_buf[fs->ate_size];
+	uint8_t inval_buf[fs->ate_size - INVAL_FLAG_OFFSET];
 	uint32_t addr = fs->ate_wra + fs->ate_size;
 
-	memset(inval_buf, 0, sizeof(inval_buf));
+	memset(inval_buf, fs->flash_params->erase_value, sizeof(inval_buf));
+	inval_buf[0] = INVAL_VAL;
+
 	while (addr <= (fs->offset + fs->sector_cnt * fs->sector_size) - fs->ate_size) {
-		rc = flash_write(fs->flash_dev, addr, inval_buf, sizeof(inval_buf));
+		rc = flash_write(fs->flash_dev, addr + INVAL_FLAG_OFFSET, inval_buf,
```

Comment:
It is basically what I am doing. It is impossible to write less than 4 bytes to the flash. But by creating a bitmask like this ``0x00FFFFFF` we ensure that only the invalid flag value is written, preserving the last three bytes to whatever the value was since you can not write a bit to 1 if it previously has been written to a 0 in flash.